src/components/Grid/Grid.tsx

```diff
@@ -22,16 +24,75 @@ export const Grid: React.FC<GridProps> = ({
   initialItems,
   updateItems,
   gapSize,
+  activeTool,
 }) => {
   const [size, setSize] = React.useState<Size | null>();
   const [hasRendered, setHasRendered] = React.useState<boolean>(false);
   const [items, setItems] = React.useState<Array<TopicMapItem>>(initialItems);
+  const [boxStartPosition, setBoxStartPosition] = React.useState<number | null>(
+    null,
+  );
 
   const elementRef = React.useRef<HTMLDivElement>(null);
 
   /* TODO: Translate */
   const gridIndicatorLabel = "Click to create a new element";
 
+  const createBoxStart = React.useCallback((index: number) => {
+    setBoxStartPosition(index);
+  }, []);
+
+  const createBoxEnd = React.useCallback(
+    (indicatorIndex: number) => {
+      if (activeTool === ToolbarButtons.CreateBox) {
+        if (!boxStartPosition && boxStartPosition !== 0) {
+          throw new Error("Box start position is not defined.");
+        }
+
+        // Get x and y percentage position
+        const x = boxStartPosition % numberOfColumns;
+        const y = Math.floor(boxStartPosition / numberOfColumns);
+
+        const xPercentagePosition = (x / numberOfColumns) * 100;
+        const yPercentagePosition = (y / numberOfRows) * 100;
+
+        // Get height percentage
+        const yEnd = Math.floor(indicatorIndex / numberOfColumns);
+        const yEndPercentagePosition = ((yEnd + 1) / numberOfRows) * 100;
+
+        const heightPercentage = yEndPercentagePosition - yPercentagePosition;
+
+        // Get width percentage
+        const indicatorValue = indicatorIndex + 1;
+        const lastIndexOnColumn = indicatorValue % numberOfColumns === 0;
+
+        const xEnd = indicatorValue % numberOfColumns;
+        const xEndPercentagePosition = lastIndexOnColumn
+          ? 100
+          : (xEnd / numberOfColumns) * 100;
+
+        const widthPercentage = xEndPercentagePosition - xPercentagePosition;
+
+        // Create box
+        const id = (items.length + 1).toString();
+
+        const newItem = {
+          id: items ? id : "1",
```

Comment:
Is the `items` null check necessary?

```suggestion
          id,
```

Reply:
No, works with your suggestion 👍